csharp/src/Microsoft.ML.OnnxRuntime/Tensors/SparseTensor.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// This file is copied and adapted from the following git repository -
+// https://github.com/dotnet/corefx
+// Commit ID: bdd0814360d4c3a58860919f292a306242f27da1
+// Path: /src/System.Numerics.Tensors/src/System/Numerics/Tensors/SparseTensor.cs
+// Original license statement below -
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System;
+
+namespace Microsoft.ML.OnnxRuntime.Tensors
+{
+    /// <summary>
+    /// Represents a multi-dimensional collection of objects of type T that can be accessed by indices.  Unlike other Tensor&lt;T&gt; implementations SparseTensor&lt;T&gt; does not expose its backing storage.  It is meant as an intermediate to be used to build other Tensors, such as CompressedSparseTensor.  Unlike CompressedSparseTensor where insertions are O(n), insertions to SparseTensor&lt;T&gt; are nominally O(1).
+    /// </summary>
+    /// <typeparam name="T">type contained within the Tensor.  Typically a value type such as int, double, float, etc.</typeparam>
+    internal class SparseTensor<T> : Tensor<T>
```

Comment:
Same comment as CompressedSparseTensor - you can always get these classes in the future if you need them. I would just leave them out completely until they are needed.